sonarts-sq-plugin/sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ContextualServerTest.java

```diff
@@ -43,20 +46,34 @@
 
   @Rule
   public final LogTester logTester = new LogTester();
+  private ExternalProcessStreamConsumer externalProcessStreamConsumer;
+
+  @Before
+  public void setUp() {
+    externalProcessStreamConsumer = new ExternalProcessStreamConsumer();
+    externalProcessStreamConsumer.start();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    externalProcessStreamConsumer.stop();
+  }
 
   @Test
-  public void should_start_and_stop() {
+  public void should_start_and_stop() throws Exception {
     ContextualServer contextualServer = getContextualServer();
     contextualServer.start();
     assertThat(contextualServer.isAlive()).isTrue();
-    assertThat(logTester.logs()).contains("SonarTS Server is started");
+    Thread.sleep(2_000);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-443&issues=AWEiyOv91DTsLFfQjB92&open=AWEiyOv91DTsLFfQjB92)

[ISSUE_KEY]: AWEiyOv91DTsLFfQjB92

Reply:
This issue has been resolved by a user in SonarQube

[ISSUE_STATUS]: RESOLVED